api/src/main/java/org/apache/cloudstack/api/response/IPAddressResponse.java

```diff
@@ -159,10 +159,9 @@ public class IPAddressResponse extends BaseResponseWithAnnotations implements Co
     @Param(description="the name of the Network where ip belongs to")
     private String networkName;
 
-    /*
-        @SerializedName(ApiConstants.JOB_ID) @Param(description="shows the current pending asynchronous job ID. This tag is not returned if no current pending jobs are acting on the volume")
-        private IdentityProxy jobId = new IdentityProxy("async_job");
-    */
+    @SerializedName(ApiConstants.HAS_RULES)
+    @Param(description="whether the ip address has Firewall/PortForwarding/LoadBalancing rules defined")
```

Comment:
please help me understand where this is used or how is it helpful. Also I see we are currently setting this flag only based on firewall rules (not portforwarding/loadbalancing rules), do we need to update the comment !

ipResponse.setHasRules(firewallRulesDao.countRulesByIpId(ipAddr.getId()) > 0);

Reply:
This is used to determine if the "make source nat"button should be activated.

(I think it is base on all rules, independent of type. I'll have a look to see if I missed something.

Reply:
@harikrishna-patnala firewall rules contains a rule type and will store all rules (also for portforwarding and loadbalancing) This should work.